seed/models/columns.py

```diff
@@ -982,22 +982,30 @@ def select_col_obj(column_name, table_name, organization_column):
                     organization=organization,
                     table_name__in=[None, ''],
                     column_name=field['from_field'],
-                    units_pint=field.get('from_units'),  # might be None
                     is_extra_data=False  # data from header rows in the files are NEVER extra data
                 )
             except Column.MultipleObjectsReturned:
                 _log.debug(
                     "More than one from_column found for {}.{}".format(field['to_table_name'],
                                                                        field['to_field']))
 
-                # TODO: write something to remove the duplicate columns
-                from_org_col = Column.objects.filter(organization=organization,
-                                                     table_name__in=[None, ''],
-                                                     column_name=field['from_field'],
-                                                     units_pint=field.get('from_units'),
-                                                     # might be None
-                                                     is_extra_data=is_extra_data).first()
-                _log.debug("Grabbing the first from_column")
+                all_from_cols = Column.objects.filter(
+                    organization=organization,
+                    table_name__in=[None, ''],
+                    column_name=field['from_field'],
+                    is_extra_data=False
+                )
+
+                ColumnMapping.objects.filter(column_raw__id__in=models.Subquery(all_from_cols.values('id'))).delete()
+                all_from_cols.delete()
+
+                from_org_col, _ = Column.objects.get_or_create(
+                    organization=organization,
+                    table_name__in=[None, ''],
+                    column_name=field['from_field'],
+                    is_extra_data=False  # data from header rows in the files are NEVER extra data
```

Comment:
Is it true that data from header rows are never extra data? I thought I could make up an arbitrary column in my csv/excel file and map it as extra data

Reply:
The type of Column object this is referring to is a representation of the "raw" header row we use for the ColumnMapping model. The ColumnMapping model takes a "PropertyState" or "TaxLotState" Column object (see `table_name` attribute), and associates that to one of these "raw" Column objects to create the mapping.

The comment could be improved, so I'll do that for this line and the original line in the `try` block above here.